pkg/sql/colexec/agg/variance/variance.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package Variance
+
+import (
+	"math"
+	"reflect"
+	"strconv"
+	"unsafe"
+
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+)
+
+func ReturnType(_ []types.Type) types.Type {
+	return types.New(types.T_float64, 0, 0, 0)
+}
+
+func String(b []byte) (s string) {
+	pbytes := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+	pstring := (*reflect.StringHeader)(unsafe.Pointer(&s))
+	pstring.Data = pbytes.Data
+	pstring.Len = pbytes.Len
+	return
+}
+
+/*
+	case types.Decimal64:
+		res, _ := strconv.ParseFloat(String(v.Decimal64ToString(variance.inputType.Scale)), 64)
+		return res
+	case types.Decimal128:
+		res, _ := strconv.ParseFloat(String(v.Decimal128ToString(variance.inputType.Scale)), 64)
+*/
+
+//New1 is used to Created a Variance which supports float,int,uint
+func New1[T1 types.Floats | types.Ints | types.UInts]() *Variance[T1] {
+	return &Variance[T1]{}
+}
+
+//New2 is used to Created a Variance which supports decimal64
+//if is decimal,you need to give pricision
+func New2() *Variance2 {
+	return &Variance2{inputType: types.New(types.T_decimal64, 0, 0, 0)}
+}
+
+//New2 is used to Created a Variance which supports decimal64
+//if is decimal,you need to give pricision
+func New3() *Variance3 {
+	return &Variance3{inputType: types.New(types.T_decimal64, 0, 0, 0)}
+}
+
+func (variance *Variance[T1]) Grows(sizes int) {
+	if len(variance.avgs) == 0 {
+		variance.avgs = make([]float64, sizes)
+		variance.sizes = make([]float64, sizes)
+	} else {
+		for i := 0; i < sizes; i++ {
+			var a float64
+			variance.avgs = append(variance.avgs, a)
+			variance.sizes = append(variance.sizes, 0)
+		}
+	}
+}
+
+func (variance *Variance[T1]) Eval(vs []float64) []float64 {
+	for i, v := range vs {
```

Comment:
I am still confused.   Let me ask this, for the following query and a huge table t

select variance(x) from t;

You should only use 24 bytes of memory, is this right?

Reply:
confused? why 24 bytes? the vs is uesd to store the final result for every group.

Reply:
> confused? why 24 bytes? the vs is used to store the final result for every group.

Reply:
There is no group by clause, so it is just one group.    You are storing 3 numbers, 8 bytes each, so 24.   Correct?